File: 217.Contains_Duplicate.js
// Solution: #1
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    let uniqeNumbers = new Set();

    for (const n of nums) {
        if (uniqeNumbers.has(n))
            return true;
        uniqeNumbers.add(n);

    }

    return false;
};


// Solution: #2
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    let uniqeNumbers = new Set(nums);

    if (nums.length === uniqeNumbers.size) 
        return false;

    else
        return true;

};

File: DSA/217.Contains_Duplicate.cpp
using namespace std;
#include <bits/stdc++.h>

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        map<int, int> nums_map;

        for (int i = 0; i < nums.size(); i++) {
            nums_map[nums[i]]++;

            if (nums_map[nums[i]] > 1) {
                return (true);
            }
        }

        return (false);
    }
};
